contracts/RMRK/extension/tokenProperties/IRMRKPropertiesAccessControl.sol

```diff
@@ -0,0 +1,157 @@
+// SPDX-LicenseIdentifier: Apache-2.0
+
+pragma solidity ^0.8.18;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+
+import "../../library/RMRKErrors.sol";
+
+interface IRMRKPropertiesAccessControl {
+    /**
+     * @notice A list of supported parameter types.
+     * @return The `string` type.
+     * @return The `address` type.
+     * @return The `bytes` type.
+     * @return The `uint` type.
+     * @return The `bool` type.
+     */
+    enum ParameterType {
+        STRING,
+        ADDRESS,
+        BYTES,
+        UINT,
+        BOOL
+    }
+
+    /**
+     * @notice A list of supported access types.
+     * @return The `Issuer` type, where only the issuer can manage the parameter.
+     * @return The `Collaborator` type, where only the collaborators can manage the parameter.
+     * @return The `IssuerOrCollaborator` type, where only the issuer or collaborators can manage the parameter.
+     * @return The `TokenOwner` type, where only the token owner can manage the parameters of their tokens.
+     * @return The `SpecificAddress` type, where only specific addresses can manage the parameter.
+     */
+    enum AccessType {
+        Issuer,
+        Collaborator,
+        IssuerOrCollaborator,
+        TokenOwner,
+        SpecificAddress
+    }
+
+    /**
+     * @notice Used to noitfy listeners that the access control settings for a specific parameter have been updated.
+     * @param collection Address of the collection.
+     * @param parameterType The type of parameter for which the access control settings have been updated.
+     * @param parameterId The ID of the parameter for which the access control settings have been updated.
```

Comment:
What's this Id? Is it some mapping from the property name? If that's the case, can't we use the name directly?

Reply:
I later realized it was the id from keyToId we use internally in token properties. I don't think the user needs to know about it.

Reply:
Yeah, I think this is better than using keys as it is a fixed size variable. We need to figure out how to package this for the proposal.. 🤔

I'd leave the documentation as it explains how we are approaching the access control if someone with sufficient knowledge wants to verify the operation